x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/persistence/ChunkedTrainedModelRestorer.java

```diff
@@ -201,6 +207,24 @@ private void doSearch(
         }
     }
 
+    static SearchResponse retryingSearch(Client client, SearchRequest searchRequest, int retries, TimeValue sleep)
+        throws InterruptedException {
+        int failureCount = 0;
+
+        while (true) {
+            try {
+                return client.search(searchRequest).actionGet();
+            } catch (SearchPhaseExecutionException e) {
```

Comment:
It's highly likely that this will be wrapped in a `RemoteTransportException`, so catching it like this won't work.

I think you should catch `Exception` and then use something like:

```
    if (ExceptionsHelper.unwrapCause(e) instanceof SearchPhaseExecutionException) {
        // do the processing
    } else {
        throw e;
    }
```

Reply:
I think `ExceptionsHelper.unwrapCause(e) instanceof CircuitBreakingException` should also be accepted as a reason to retry. I'm not clear if the `CircuitBreakingException` will always be wrapped in a `SearchPhaseExecutionException`. It seems safer to accept either as a reason to retry.